nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/search/attributematchers/AttributeMatcher.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.web.search.attributematchers;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.nifi.web.search.query.SearchQuery;
+
+import java.util.List;
+
+public interface AttributeMatcher<T> {
+    String SEPARATOR = ": ";
+
+    void match(T component, SearchQuery query, List<String> matches);
+
+    static void addIfMatching(final String searchTerm, final String subject, final String label, final List<String> matches) {
+        final String match = (label == null) //
+                ? subject //
+                : new StringBuilder(label).append(SEPARATOR).append(subject).toString();
```

Comment:
`label + ": " + subject` is perfectly fine, `StringBuilder` will be used automatically under the hood.
Its explicit use becomes important when used in a separate block (like in a `for` cycle for example).